fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PermissiveAvailabilitySpec.groovy

```diff
@@ -0,0 +1,145 @@
+        given:
+        interval1 = new Interval('2010-01-01/2020-12-31')
+        interval2 = new Interval('2017-01-01/2020-12-31')
+
+        permissiveAvailability = new PermissiveAvailability(
+                tableName,
+                [column1, column2] as Set,
+                new TestDataSourceMetadataService([
+                        (column1): [interval1] as Set,
+                        (column2): [interval2] as Set
+                ])
+        )
+
+        expect:
+        permissiveAvailability.getAllAvailableIntervals() == [
+                (column1): [interval1] as Set,
+                (column2): [interval2] as Set
+        ] as LinkedHashMap
+    }
+
+    @Unroll
+    def "getAvailableIntervals returns the union of requested columns with available intervals - #firstInterval & #secondInterval"() {
+        given:
+        interval1 = new Interval(firstInterval)
+        interval2 = new Interval(secondInterval)
+
+        permissiveAvailability = new PermissiveAvailability(
+                tableName,
+                [column1, column2] as Set,
+                new TestDataSourceMetadataService([
+                        (column1): [interval1] as Set,
+                        (column2): [interval2] as Set
+                ])
+        )
+
+        DataSourceConstraint dataSourceConstraint = Mock(DataSourceConstraint)
+        dataSourceConstraint.getAllColumnNames() >> ['column1', 'column2']
+
+        expect:
+        permissiveAvailability.getAvailableIntervals(dataSourceConstraint) == new SimplifiedIntervalList(
+                expected.stream()
+                        .map{it -> new Interval(it)}
+                        .collect(Collectors.toSet())
+        )
+
+        where:
+        firstInterval           | secondInterval          | expected
+        '2017-01-01/2017-02-01' | '2017-01-01/2017-02-01' | ['2017-01-01/2017-02-01']                          // full overlap (start/end, start/end)
+        '2017-01-01/2017-02-01' | '2018-01-01/2018-02-01' | ['2017-01-01/2017-02-01', '2018-01-01/2018-02-01'] // 0 overlap (-10/-1, 0/10)
+        '2017-01-01/2017-02-01' | '2017-02-01/2017-03-01' | ['2017-01-01/2017-03-01']                          // 0 overlap abutting (-10/0, 0/10)
+        '2017-01-01/2017-02-01' | '2017-01-15/2017-03-01' | ['2017-01-01/2017-03-01']                          // partial front overlap (0/10, 5/15)
+        '2017-01-01/2017-02-01' | '2016-10-01/2017-01-15' | ['2016-10-01/2017-02-01']                          // partial back overlap (0/10, -5/5)
+        '2017-01-01/2017-02-01' | '2017-01-15/2017-02-01' | ['2017-01-01/2017-02-01']                          // full front overlap (0/10, 5/10)
+        '2017-01-01/2017-02-01' | '2017-01-01/2017-01-15' | ['2017-01-01/2017-02-01']                          // full back overlap (0/10, 0/5)
+        '2017-01-01/2017-02-01' | '2017-01-15/2017-01-25' | ['2017-01-01/2017-02-01']                          // fully contain (0/10, 3/9)
```

Comment:
One way to do it would be like this:

```groovy

Bar mockBar = Mock(Bar)

@Unroll
def "The #thing always does #actionCount #action whenever it is poked and #reason"() {
    given: "A configured thing and a mock"
    thing."set$initialParam"(initialState)

    when: "It gets poked"
    thing.poke()

    then: "It has taken the action the correct number of times"
    actionCount * mockBar."action"()

    where:
    thing     | actionCount | action | initialParam | initialState | reason
    new Dog() | 1           | "jump" | "Barking"    | true         | "it is barking"
}
```

This example is a bit fancy, and shows off much more than how to get a string to unroll as part of the test name, but you can see that the `reason` `where` variable is _only_ used as an unrolled parameter in the test name.

In a nutshell, you can put a `#` in front of any `where` parameter in the test name when using `@Unroll`, and the value of that variable will be swapped into the test name when Spock generates the tests (which is on for each iteration in the `where` block.